virtme/commands/run.py

```diff
@@ -1840,6 +1876,30 @@ def get_net_mac(index):
     # Load a normal kernel
     qemuargs.extend(["-kernel", kernel.kimg])
     if kernelargs:
+        if args.systemd:
+            init_environment_vars = []
+            for arg in kernelargs:
+                match = re.match(r"(virtme_.*)=(.*)", arg)
+                if not match:
+                    continue
+                init_environment_vars.append(
+                    shlex.quote(f"{match.group(1)}={match.group(2)}")
```

Comment:
```suggestion
                    f"{match.group(1)}={match.group(2)}"
```
should be enough. The `shlex.join` handles the quoting for us.

Reply:
Indeed, done in https://github.com/arighi/virtme-ng/pull/280/commits/6207c487e808843d57fcfba1508fb581a1024f55